flag.go

```diff
@@ -0,0 +1,192 @@
+package ldclient
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"io"
+	"reflect"
+	"strconv"
+)
+
+type FeatureFlag struct {
+	Name         string      `json:"name" bson:"name"`
+	Key          string      `json:"key" bson:"key"`
+	Version      int         `json:"version" bson:"version"`
+	On           bool        `json:"on" bson:"on"`
+	Salt         string      `json:"salt" bson:"salt"`
+	Sel          string      `json:"sel" bson:"sel"`
+	Targets      []Target    `json:"targets" bson:"targets"`
+	Rules        []Rule      `json:"rules" bson:"rules"`
+	Fallthrough  Rule        `json:"fallthrough" bson:"fallthrough"`
+	OffVariation interface{} `json:"offVariation" bson:"offVariation"`
+	Archived     bool        `json:"archived" bson:"archived"`
+}
+
+// Expresses a set of AND-ed matching conditions for a user, along with
```

Comment:
comment on exported type Rule should be of the form "Rule ..." (with optional leading article)